crates/cargo-util/src/process_builder.rs

```diff
@@ -363,46 +510,80 @@ impl ProcessBuilder {
     /// let cmd = cmd.wrapped(Some("sccache"));
     /// ```
     pub fn wrapped(mut self, wrapper: Option<impl AsRef<OsStr>>) -> Self {
-        let wrapper = if let Some(wrapper) = wrapper.as_ref() {
-            wrapper.as_ref()
-        } else {
-            return self;
-        };
-
-        if wrapper.is_empty() {
-            return self;
+        if let Some(wrapper) = wrapper.as_ref() {
+            let wrapper = wrapper.as_ref();
+            if !wrapper.is_empty() {
+                self.wrappers.push(wrapper.to_os_string());
+            }
         }
+        self
+    }
+}
 
-        let args = once(self.program).chain(self.args.into_iter()).collect();
+/// Forces the command to use `@path` argfile.
+///
+/// You should set `__CARGO_TEST_FORCE_ARGFILE` to enable this.
+fn debug_force_argfile(retry_enabled: bool) -> bool {
+    cfg!(debug_assertions) && env::var("__CARGO_TEST_FORCE_ARGFILE").is_ok() && retry_enabled
+}
 
```

Comment:
Observation: I could see in the future providing a flag in release mode as a way for people to workaround or debug problems.